### ISSUE TITLE
How to expand Strlen when i use tracepoint:cgroup:cgroup_rmdir

### ISSUE THREAD

**Issue by mYu4N** [bpftrace/bpftrace#3617]
# BPFTRACE_MAX_STRLEN=1024 bpftrace  cg.bt
error: <unknown>:0:0: in function tracepoint_cgroup_cgroup_rmdir_1 i64 (ptr): Looks like the BPF stack limit is exceeded. Please move large on stack variables into BPF per-cpu array map. For non-kernel uses, the stack can be increased using -mllvm -bpf-stack-size.

# bpftrace -V
bpftrace v0.21.2

tracepoint:cgroup:cgroup_rmdir
{
    time("%Y-%m-%d %H:%M:%S ");
    printf (" %s root:%d path:%s\n", probe, args->root, str(args->path));
    printf ("%d %d %s %s %s \n", pid, curtask->parent->pid, comm, curtask->parent->comm, ustack)
}

only can be set 110 # BPFTRACE_MAX_STRLEN=110 bpftrace  cg.bt
Attaching 1 probe...

But the cgroup path is incomplete，look this pic ,it cannot print cri-containerd-xxxx.scope

 
<img width="1458" alt="image" src="https://github.com/user-attachments/assets/9e0c3d25-e2db-4ea1-a57e-584720543e4e">

`2024-12-02 11:02:07  tracepoint:cgroup:cgroup_rmdir root:5 path:/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod3fa9da2e_096a_4ff5_89a2_b8cbf85e7d3e.slice/cri..   **This has been omitted here**
52760 52518 runc containerd-shim
        runtime/internal/syscall.Syscall6+14
        syscall.Syscall+70
        syscall.Syscall.abi0+48
        github.com/opencontainers/runc/vendor/golang.org/x/sys/unix.Unlinkat+133
        github.com/opencontainers/runc/libcontainer/cgroups.rmdir+52
        github.com/opencontainers/runc/libcontainer/cgroups.RemovePath+39
        github.com/opencontainers/runc/libcontainer/cgroups.RemovePaths+363
        github.com/opencontainers/runc/libcontainer/cgroups/systemd.(*legacyManager).Destroy+312
        github.com/opencontainers/runc/libcontainer.destroy+322
        github.com/opencontainers/runc/libcontainer.(*stoppedState).destroy+22
        github.com/opencontainers/runc/libcontainer.(*linuxContainer).Destroy+154
        main.destroy+34
        main.glob..func3+796
        github.com/opencontainers/runc/vendor/github.com/urfave/cli.HandleAction+112
        github.com/opencontainers/runc/vendor/github.com/urfave/cli.Command.Run+1598
        github.com/opencontainers/runc/vendor/github.com/urfave/cli.(*App).Run+2919
        main.main+4646
        runtime.main+722
        runtime.goexit.abi0+1`

**Comment by jordalgo** [comment]
@mYu4N Do you mind trying to replicate this on a master build of bpftrace? There were a lot of fixes for big strings/objects that have not yet been released (we hope to do a release in the next two weeks with these changes).

**Comment by danobi** [comment]
Yep, we finalized support for big strings on master. BPFTRACE_MAX_STRLEN can go as high as you want now (up to 32k)
